python/src/KernelSmoothing_doc.i.in

```diff
@@ -335,7 +391,96 @@ automaticLowerBound : bool
 Parameters
 ----------
 automaticUpperBound : bool
-    Flag to tell if the user-defined upper bound has to be taken into account (value *False*) or if the maximum of the given sample has to be used (value *True*)."
+    Flag to tell if the upper bound is automatically calculated from the sample.
+
+Notes
+-----
+This treatment is available in dimension 1 only.
+The automatic upper bound is the maximum of the given sample. In the other case,
+the user has to specify the upper bound."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::KernelSmoothing::getBinning
+"Accessor to the binning flag.
+
+Returns
+-------
+binning : bool
+    Flag to tell if the binning treatment is activated.
+
+Notes
+-----
+This treatment is available in dimension 1 and 2 only."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::KernelSmoothing::setBinning
+"Accessor to the binning flag.
+
+Parameters
+----------
+binning : bool
+    Flag to tell if the binning treatment is activated.
+
+Notes
+-----
+This treatment is available in dimension 1 and 2 only.
+It creates a regular grid of *binNumber*
+intervals in each
+dimension, then the unit weight of each point is linearly affected to the vertices
+of the bin containing the point (see [wand1994]_ appendix D, page 182).
+The `KernelSmoothing-BinNumber` key of the class  :class:`~openturns.ResourceMap` defines the default value of the 
+number of bins used in the _binning_ algorithm to improve the evaluation speed."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::KernelSmoothing::setUseLogTransform
+"Accessor to the log-transform flag.
+
+Parameters
+----------
+useLogTransform : bool
+    Flag to tell if the kernel smoothing distribution is built on the log-transformed data.
+
+Notes
+-----
+This treatment is available in dimension 1 only. See [charpentier2015]_ to get more details.
+
+We denote by :math:`(X_i)_{1 \leq i  \leq n}`
+some independent random variates, identically distributed according to :math:`X`.
+
+The log-transform treatment maps each :math:`X_j` into :math:`Y_j` as follows:
+
+.. math:: 
+
+    Y_j  = T(X_j) = \left |
+    \begin{array}{ll}
+       \log (X_j - \min_{i} X_i + \delta) & \mbox{if } \gamma_1(X) >0\\
+       \log (\max_{i} X_i - X_j + \delta) & \mbox{if } \gamma_1(X) >0
+       \end{array}
+       \right.
```

Comment:
Please avoid to duplicate these equations, which are already presented in python/doc/theory/data_analysis/kernel_smoothing.rst.

Reply:
I replaced these lines by a link to the theory section of kernel smoothing. I kept the line describing how the shift is fixed using ResourceMap.